src/automaton/core/interop/ethereum/eth_helper_functions.h

```diff
@@ -125,7 +131,437 @@ static status eth_getTransactionReceipt(const std::string& url, const std::strin
   return curl_post(url, ss.str());
 }
 
+// Argument encoding and decoding
+
+static const uint32_t BUFFER_SIZE = 2048;
+
+enum abi_type {
+  empty = 0,
+  numerical = 1,  // bool, int<>, uint<>, fixed, unfixed, fixed<M>x<N>, ufixed<M>x<N>
+  fixed_size_bytes = 2,  // bytes<>, address, function
+  dynamic_size_bytes = 4,  // bytes, string
+  fixed_size_array = 8,  // uint256[3], bytes[7]
+  dynamic_size_array = 16  // bool[], bytes[], uint256[][]
+};
 
+std::ostream& operator<<(std::ostream& os, const abi_type& obj) {
+  os << (int)obj << ": ";  // NOLINT
+
+  if (obj & numerical) {
```

Comment:
You are not passing in a mask, but rather an enum. Why not use a switch here?

Reply:
Because obj can be in two categories, combination of 2 values, and if switch is used, there will be more cases and will be more complicated, but it is an option